projects/plugins/jetpack/class.jetpack.php

```diff
@@ -4362,15 +4362,19 @@ function ( $domains ) {
 						exit;
 					}
 
-					$redirect_url = self::admin_url(
-						array(
-							'page'     => 'jetpack',
-							'action'   => 'authorize_redirect',
-							'dest_url' => rawurlencode( $dest_url ),
-							'done'     => '1',
-						)
+					$redirect_args = array(
+						'page'     => 'jetpack',
+						'action'   => 'authorize_redirect',
+						'dest_url' => rawurlencode( $dest_url ),
+						'done'     => '1',
 					);
 
+					if ( ! empty( $_GET['from'] ) && 'jetpack_site_only_checkout' === $_GET['from'] ) {
+						$redirect_args['from'] = 'jetpack_site_only_checkout';
+					}
+
+					$redirect_url = self::admin_url( $redirect_args );
+
 					wp_safe_redirect( static::build_authorize_url( $redirect_url ) );
```

Comment:
```suggestion
					wp_safe_redirect( static::build_authorize_url( $self::admin_url( $redirect_url ) ) );
```
This is just an opinion, but I think we could get rid of the intermediary variable `$redirect_url` in here.